metrics/src/cow.rs

```diff
@@ -178,6 +178,35 @@ where
 
         T::owned_from_parts(cow.ptr, &cow.metadata)
     }
+
+    /// Extracts the borrowed data.
+    ///
+    /// Returns `None` if the data is either shared or owned.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use metrics::SharedString;
+    ///
+    /// let s = SharedString::from_borrowed("foobar");
+    /// assert_eq!(s.as_borrowed(), Some("foobar"));
+    ///
+    /// let s = SharedString::from_owned("foobar".to_owned());
+    /// assert_eq!(s.as_borrowed(), None);
+    ///
+    /// let s = SharedString::from_shared("foobar".to_owned().into());
+    /// assert_eq!(s.as_borrowed(), None);
+    /// ```
+    pub fn as_borrowed(&self) -> Option<&'a T> {
+        match self.metadata.kind() {
+            Kind::Owned | Kind::Shared => None,
+            Kind::Borrowed => {
+                // SAFETY: We know the contained data is borrowed from 'a, we're simply
+                // restoring the original immutable reference and returning a copy of it.
+                Some(unsafe { &*T::borrowed_from_parts(self.ptr, &self.metadata) })
+            }
+        }
+    }
```

Comment:
Unless there's a compelling reason to have this be a standalone method, we should just move the logic down into `impl From<Cow<'a, T>> for std::borrow::Cow<'a, T>`.

Reply:
The problem with that is, the `From` impl always requires a `Clone` (or taking ownership, but the way the `SharedString` is exposed it is usually through a reference), the `as_borrowed()` allows for a decision.